system/jlib/jbuff.hpp

```diff
@@ -57,48 +57,67 @@ class jlib_decl MemoryAttr
 
 //--------------------------------------------------------------------------------------------------------------------
 
-template <class CLASS> class OwnedMalloc
+template <class CLASS, void (*FREE_FUNC)(CLASS *)> class OwnedPtrCustomFree
 {
+    typedef OwnedPtrCustomFree<CLASS, FREE_FUNC> SELF;
+protected:
+    CLASS *ptr = nullptr;
+
 public:
-    inline OwnedMalloc()                        { ptr = NULL; }
-    inline OwnedMalloc(CLASS * _ptr)            { ptr = _ptr; }
-    explicit inline OwnedMalloc(unsigned n, bool clearMemory = false) { doAllocate(n, clearMemory); }
-    inline ~OwnedMalloc()                       { free(ptr); }
+    OwnedPtrCustomFree<CLASS, FREE_FUNC>() { }
+    OwnedPtrCustomFree<CLASS, FREE_FUNC>(CLASS *_ptr) : ptr(_ptr) { }
+    OwnedPtrCustomFree<CLASS, FREE_FUNC>(SELF &&_ptr) { ptr = _ptr.getClear(); }
+    ~OwnedPtrCustomFree<CLASS, FREE_FUNC>() { FREE_FUNC(ptr); }
```

Comment:
Can FREE_FUNC be called on a null pointer?  If not, this call needs a check.  If yes, then checks in operator = and setown can be removed.  

delete and free do not need to check.  I think it may also be true of the EVP functions, but this class should be consistent.